ingress/controllers/nginx/configuration.md

```diff
@@ -0,0 +1,394 @@
+
+`ingress.kubernetes.io/limit-rps`: number of allowed connections per second from a single IP address
+
+
+Is possible to specify both annotation in the same Ingress rule. In this case the order of the limit is `limit-connections` and then `limit-rps`
+
+
+### Secure NGINX upstreams
+
+By default NGINX uses `http` to reach the services. Adding the annotation `ingress.kubernetes.io/secure-upstream: "true"` in the ingress rule changes the protocol to `https`.
+
+
+### Whitelist source range
+
+Using the annotation `ingress.kubernetes.io/whitelist-source-range` is possible to specify one or ranges of client IP addresses from which is possible to access the `Path` in the Ingress rule, e.g. `10.0.0.0/24,172.10.0.1`
+For a global restriction (any URL) is possible to use `whitelist-source-range` in the NGINX config map
+
+*Note:* adding an annotation overrides any global restriction
+
+Please check the [whitelist](examples/whitelist/README.md) example
+
+
+
+### **Allowed parameters in configuration config map:**
+
+##### *body-size:*
+
+Sets the maximum allowed size of the client request body. See NGINX [client_max_body_size](http://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size)
+
+
+##### *custom-http-errors:*
+
+Enables which HTTP codes should be passed for processing with the [error_page directive](http://nginx.org/en/docs/http/ngx_http_core_module.html#error_page)
+Setting at least one code this also enables [proxy_intercept_errors](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_intercept_errors) (required to process error_page)
+For instance setting `custom-http-errors: 404,415` 
+
+
+##### *enable-sticky-sessions:* 
+
+Enables sticky sessions using cookies. This is provided by [nginx-sticky-module-ng](https://bitbucket.org/nginx-goodies/nginx-sticky-module-ng) module
+
+
+##### *enable-vts-status:*
+
+Allows the replacement of the default status page with a third party module named [nginx-module-vts](https://github.com/vozlt/nginx-module-vts)
+
+
+##### *error-log-level:*
+
+Configures the logging level of errors. Log levels above are listed in the order of increasing severity
+http://nginx.org/en/docs/ngx_core_module.html#error_log
+
+
+##### *retry-non-idempotent:*
+
+Since 1.9.13 NGINX will not retry non-idempotent requests (POST, LOCK, PATCH) in case of an error in the upstream server. 
+The previous behavior can be restored using the value "true"
+
+
+##### *hsts:*
+
+Enables or disables the header HSTS in servers running SSL.
+HTTP Strict Transport Security (often abbreviated as HSTS) is a security feature (HTTP header) that tell browsers that it should only be communicated with using HTTPS, instead of using HTTP.
+https://developer.mozilla.org/en-US/docs/Web/Security/HTTP_strict_transport_security
+Why HSTS is important?
+
+
+##### *hsts-include-subdomains:*
+
+Enables or disables the use of HSTS in all the subdomains of the servername
```

Comment:
can you give a simple example with 2 subdomains and how this forces https on one and not the other when the setting is not present?